### ISSUE TITLE
[gvar] Move low-level routines to a common place

### ISSUE THREAD

**Issue by brawer** [fonttools/fonttools#583]
Some of the fonttools code for compiling and decompiling variation data structures (tuples, packed deltas, etc.) could probably be moved from the `gvar` table into otTables or another base library. After that, implementing `cvar` would be easier.

Generally, looking at my old `gvar` code, I’m not terribly happy with it. While the TTX output is easy to read for humans, it also hides a fair bunch of internals. For example, one does not see which tuples are shared versus embedded. I think the code in ttLib should become more “assembler” than “compiler”, and the higher-level stuff should move to varLib.

**Comment by twardoch** [comment]
Ideally, the ttx format (not just for gvar but overall) might be a bit simplified and perhaps hide some of the gory internals while ttLib could expose more lower-level stuff. 

When researching this [curious Apple TT behavior](https://github.com/twardoch/test-fonts/blob/master/varia/160413-EvenOddTT/README.md) I tried to set the bit 6 of the flags field of a glyf entry using ttLib, one that is officially "reserved" but is mentioned in an old AAT-related document, and failed, because ttLib had no mechanism to access the flags bitmask. Ultimately, I had to revert to a hex editor and calculate the offsets manually. :)
